nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
This PR needs a **reliable** (works with package.json files that aren't perfectly spaced/formatted) and **extensible** (if npm changes the "engines" field again, this code would need to be adjusted) way to retrieve the semver expression from the package.json. After a little research, it looks like the only options are:

1. implementing a full blown industrial json parser in bash just to use for this one edge case (reliable and extensible but too much code)
2. hacking together a regular expression that interprets a nested value (engines.node) in the package.json file (not reliable and not easily extensible)
3. imposing a dependency (https://stedolan.github.io/jq/)
  
4. using node  
I agree that ideally, this should work without node but given the other options, it looks like using node might be the best choice with the only caveat that it wouldn't work for the 1% of use cases where nvm have no nodes installed.